pkg/prom/metrics.go

```diff
@@ -104,3 +104,17 @@ func KubeAnnotationsToLabels(labels map[string]string) ([]string, []string) {
 func SanitizeLabelName(s string) string {
 	return invalidLabelCharRE.ReplaceAllString(s, "_")
 }
+
+// SanitizeLabels sanitizes all label names in the given map. This may cause
+// collisions, which is intentional as collisions that are not caught prior to
+// attempted emission will cause fatal errors. In the case of a collision, the
+// last value seen will be set, and all previous values will be overwritten.
+func SanitizeLabels(labels map[string]string) map[string]string {
+	response := make(map[string]string, len(labels))
+
+	for k, v := range labels {
+		response[SanitizeLabelName(k)] = v
```

Comment:
@nikovacevic What're your thoughts on adding a WRN log here? While iterating, if we see that a key already exists in the map we can log the value that is about to be overwritten?

Reply:
I'm not opposed to doing that -- but I do want to be a bit careful not to make a really spammy log, as these will get run over and over and over. So I'd probably prefer to have a diagnostic than logs, but maybe that's being too picky.

Reply:
Understood, let's leave as is for now!